libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/TracingContextPropagator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+/**
+ * Interface defining the tracing related context propagation
+ */
+public interface TracingContextPropagator {
```

Comment:
This propagator is pretty generic and should not be tight to `headers` or `spans`: it operates on `Map`s and could extract or inject whatever it needs: request headers is one example, `ThreadContextStruct::transientHeaders` is another example

Reply:
Done